cpp/src/gandiva/projector.cc

```diff
@@ -153,15 +140,32 @@ Status Projector::Make(SchemaPtr schema, const ExpressionVector& exprs,
   ARROW_RETURN_IF(configuration == nullptr,
                   Status::Invalid("Configuration cannot be null"));
 
-  // see if equivalent projector was already built
-  static Cache<ProjectorCacheKey, std::shared_ptr<Projector>> cache;
-  ProjectorCacheKey cache_key(schema, configuration, exprs, selection_vector_mode);
-  std::shared_ptr<Projector> cached_projector = cache.GetModule(cache_key);
-  if (cached_projector != nullptr) {
-    *projector = cached_projector;
-    return Status::OK();
+
+  std::shared_ptr<Cache<BaseCacheKey, std::shared_ptr<llvm::MemoryBuffer>>> shared_cache = LLVMGenerator::GetCache();
+
+
+  // Cache key ptrs to use when caching only the obj code
+  ProjectorCacheKey projector_key(schema, configuration, exprs, selection_vector_mode);
+  BaseCacheKey cache_key(projector_key, "projector");
+  std::unique_ptr<BaseCacheKey> base_cache_key = std::make_unique<BaseCacheKey>(cache_key);
+  std::shared_ptr<BaseCacheKey> shared_base_cache_key = std::move(base_cache_key);
+
+  // LLVM ObjectCache flag to use when caching only the obj code
+  bool llvm_flag = false;
+
+  std::shared_ptr<llvm::MemoryBuffer> prev_cached_obj;
+  prev_cached_obj = shared_cache->GetObjectCode(*shared_base_cache_key);
+
+  // Verify if previous projector obj code was cached
+  if(prev_cached_obj != nullptr) {
+    //ARROW_LOG(DEBUG) << "[OBJ-CACHE-LOG]: Object code WAS already cached!";
```

Comment:
Check if this comment is necessary

Reply:
This one is good for debug, I removed the // to put it back